File: ejemplos/PokeDesk/app/src/main/java/com/example/pokedesk/api/PokeApiAdapter.java
package com.example.pokedesk.api;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class PokeApiAdapter {

    private static PokeApiService API_SERVICE;
    private static final String BASE_URL = "https://pokeapi.co/api/v2/";

    public static PokeApiService getApiService() {
        final HttpLoggingInterceptor logging = new HttpLoggingInterceptor();
        logging.setLevel(HttpLoggingInterceptor.Level.BODY);
        final OkHttpClient.Builder httpClient = new OkHttpClient.Builder();
        httpClient.addInterceptor(logging);

        if (API_SERVICE == null) {
            Retrofit retrofit = new Retrofit.Builder()
                    .baseUrl(BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .client(httpClient.build())
                    .build();
            API_SERVICE = retrofit.create(PokeApiService.class);
        }
        return API_SERVICE;
    }
}

File: ejemplos/PokeDesk/app/src/main/java/com/example/pokedesk/PokeApiAdapter.kt
package com.example.pokedesk

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class PokeApiAdapter() {
    fun getRetrofit(): Retrofit {
        return Retrofit.Builder()
            .baseUrl("https://pokeapi.co/api/v2/")
            .addConverterFactory(GsonConverterFactory.create())
            .build()
    }
}